source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,343 @@
+
+An invalid connection string is the most common cause of authentication
+issues when attempting to connect to MongoDB using ``SCRAM-SHA-256``.
+
+.. tip::
+
+   For more information about connection strings,
+   see :ref:`Connection URI <node-connection-uri>` in the Connection Guide.
+
+If your connection string contains a username and password, ensure that they
+are in the correct format.
+
+.. note::
+
+   If the username or password includes any of the following characters, they
+   must be `percent encoded <https://tools.ietf.org/html/rfc3986#section-2.1>`__:
+
+   .. code-block:: none
+
+      : / ? # [ ] @
+
+   The following example shows how to percent encode "P@assword":
+
+   .. code-block:: none
+
+      password = "P%40ssword"
+
+.. _node-troubleshooting-connection-admin:
+
+Verify the User Is in the Authentication Database
+-------------------------------------------------
+
+To successfully authenticate a connection by using a username and password with
+``SCRAM-SHA-256``, the username must be defined in the authentication database.
+The default authentication database is the ``admin`` database. To use a different
+database for authentication, specify the ``authSource`` in the connection string.
+The following example instructs the driver to use ``users`` as the authentication
+database:
+
+.. code-block:: javascript
+   :copyable: true
+
+   const { MongoClient } = require("mongodb");
+   const uri = "mongodb://<username>:<password>@<hostname>:<port>/?authSource=users";
+   const client = new MongoClient(uri);
+
+You can check if this is the issue by attempting to connect to a MongoDB
+instance hosted on the local machine with the same code. A deployment on
+the same machine doesn't require any authorization to connect.
+
+Error Sending Message
+~~~~~~~~~~~~~~~~~~~~~
+
+When the driver fails to send a command after you make a request,
+it may display the following error message:
+
+.. code-block:: none
+   :copyable: false
+
+   com.mongodb.MongoSocketWriteException: Exception sending message
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+Check User Permissions
+----------------------
+
+Verify that you've accessed the MongoDB deployment with the correct user. The driver
+might generate an error message if you are using a user that does not have
+permissions to send the message.
+
+Also ensure that the user has the appropriate permissions for the message you
+are sending. MongoDB uses Role-Based Access Control (RBAC) to control access
+to a MongoDB system. For more information about how to configure RBAC in MongoDB,
+see :manual:`Default MongoDB Port </core/authorization/>`.
+
+Verify the User Is in the Authentication Database
+-------------------------------------------------
+
+Verify the user is in the correct authentication database. For more
+information about the authentication database, see
+:ref:`Authentication Error <node-troubleshooting-connection-admin>`.
```

Comment:
Suggestion:
I think the content in this section is subsumed by the info in the prior header "Check User Permissions" and therefore should be omitted. If the reader checks for the user permissions, they must find the user first.